library/common/config/config.cc

```diff
@@ -213,6 +214,8 @@ const char* config_template = R"(
           stat_prefix: hcm
           server_header_transformation: PASS_THROUGH
           stream_idle_timeout: *stream_idle_timeout
+          http2_protocol_options:
```

Comment:
@alyssawilk after reading the flow control [docs](https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/flow_control) and based on the [evidence](https://envoyproxy.slack.com/archives/CKQ2LK23G/p1635293476003300) I have I ended up with this change. The rest of it is Envoy Mobile plumbing to expose it to consumers but the real change is adding this config to the api listener. lmk what you think.

Reply:
I guess my only thought was wondering if we want a consistent default for both HTTP/1 and HTTP/2? 

Also don't you want to set the stream limit size for the upstream h2 protocol options?  I think you're setting in the hcm so "downstream"

Reply:
re: h1. I exposed the API as h2 explicit, but if we want to make it general to all http, I am not opposed.

re: hcm. Yes, only set it in the HCM because based on my reading of the code, docs, and log lines we get that is where the buffer is being blown. i.e., no individual data frame blows the watermark but the buffering that the platform bridge filter does is what blows the buffer limit in the filter manager.

Reply:
ah I hadn't realized where the buffering was problematic.  Yeah I think you're good then.